Add QC vignette and tiny change in plot_drc_4p (#99)
jpquast/protti: vignettes/quality_control_workflow.Rmd

```{r setup, message = FALSE}
library(protti)
library(magrittr)
library(dplyr)

```

After having loaded our required packages, we can create a synthetic dataset, as it should be obtained though a treatment experiment with for example a protein, metabolite or small-molecule. 
You can skip this step if you have your own data ready.

We are creating a random dataset with 100 different proteins, out of which 5 % are significantly changing upon treatment. The dataset includes 3 replicates for 2 different conditions (treated and untreated).

The `create_synthetic_data()` function is modeled after real LiP-MS data and its commonly observed data distributions.

Please note that generally, quality control should be conducted on raw unfiltered data (the direct output of your search engine of choice).

```{r create_synthetic_data}
set.seed(123) # by setting the seed we are making sure that the random object generation can be reproduced

data <- create_synthetic_data(
  n_proteins = 100,
  frac_change = 0.05,
  n_replicates = 3,
  n_conditions = 2,
  method = "random_effect",
  additional_metadata = TRUE
)

```

## Quality control

> 

Reviewer: The way I usually set up the quality control is to first look at qualitative quality (everything that does not involve intensities). Then to look at the quantitative quality. 
For the quantitative I would potentially start first with the distribution of data e.g. qc_log2_intensity_distribution. Just to see that everything looks normal and not with the CVs.

### Coefficients of variation

> A good first step to start with your quality control is to check if the coefficients of variation between your different replicates are in a reasonable range. Ideally, these should be below 15-20 %. If you see groups with higher CVs, there might have been a sample preparation error or the instrument performance was not ideal at the time of measurement. You can check if after `median_normalisation()` (depending on whether your search engine has performed median normalisation already) your CVs improve.

Reviewer: ```suggestion
A good first step of quality control is to check if coefficients of variation between replicates are in a reasonable range. Ideally, these should be below 15-20 %. If you see groups with higher CVs, there might have been a sample preparation error or the instrument performance was not ideal at the time of measurement. You can check if after `median_normalisation()` (depending on whether your search engine has performed median normalisation already) your CVs improve.
The "combined" group of CVs contains CVs across all samples and not only across the replicates of a certain condition. Ideally the combined CVs should always be the group with the highest CV.  
```

I would maybe also add clear formulas that show how the CV is calculated. E.g from intensity of a group (peptide/precursor/protein) over the replicates. and then just the formula. This will make people including Ludovic happy because they know exactly how we calculate stuff.

```{r qc_missed_cleavages, fig.width = 6, fig.height = 4, fig.align = "center"}
qc_missed_cleavages(data = data, 
                    sample = sample, 
                    grouping = peptide, 
                    missed_cleavages = n_missed_cleavage, 
                    method = "intensity",
                    intensity = raw_intensity, 
                    plot = TRUE)
```

### Sequence coverage

The following function gives you insight into the protein coverage percentages in the dataset. For a LiP-MS experiment high sequence coverages are desirable. 

To assess the distribution of the protein coverage, we are going to use the function `qc_protein_coverage()`. If you do not have a column containing the protein coverages in your data, you can use the function `sequence_coverage()` to obtain this information. 

```{r qc_sequence_coverage, fig.width = 6, fig.height = 4, fig.align = "center"}
qc_sequence_coverage(data = data, 
                     protein_identifier = protein, 
                     coverage = coverage)
``` 

### Peak width

> In order to identify potential chromatographic issues that might have occured throughout the measurement process one can take a look at the peak widths over the complete chromatogram. To do this we are going to use the function `qc_peak_width()`. This function requires either the peak start and end times or the retention time and peak width. This information can be obtained from Spectronaut (or any other search engine of your choice). 

Reviewer: ```suggestion
In order to identify potential chromatographic issues that might have occurred during the measurement we can have a look at the chromatographic median peak widths over the complete chromatogram. To do this we are going to use the function `qc_peak_width()`. This function requires either the peak start and end times or the retention time and peak width. This information can be obtained from Spectronaut (or any other search engine of your choice). 
```

The peak widths should be similar for all the measured samples of the experiment. 

```{r qc_peak_width, fig.width = 6, fig.height = 4, fig.align = "center"}
qc_peak_width(data = data,
              sample = sample, 
              intensity = peptide_intensity_missing,
              retention_time = retention_time, 
              peak_width = peak_width)
``` 

### Data completeness

> The following function `qc_data_completeness()` checks how many of all detected precursors, peptides or proteins were identified in each sample. The function uses raw intensities (not log2 transformed) and can return either a plot or a table. 

Reviewer: ```suggestion
The following function `qc_data_completeness()` checks how many of all detected precursors, peptides or proteins were identified in each sample. The function can return either a plot or a table. 
```
Actually it does not require raw intensities. should also be corrected in the function documentation. It does not matter if the intensities are transformed or not.

```{r qc_data_completeness, fig.width = 6, fig.height = 4, fig.align = "center"}
qc_data_completeness(data = data, 
                     sample = sample, 
                     grouping = peptide, 

>                      intensity = raw_intensity, 

Reviewer: ```suggestion
                     intensity = peptide_intensity_missing, 
```

                     plot = TRUE)
``` 

### Log2(intensity) distribution

For different kinds of analyses (e.g. t-tests) it is important that your data intensity follows a normal distribution. To ensure that this is the case, we are going to use the function `qc_log2_intensity_distriubution()`. The function returns a plot showing how the intensities are distributed.

```{r qc_log2_distribution, fig.width = 6, fig.height = 4, fig.align = "center"}
qc_log2_intensity_distribution(data = data,
                               sample = sample, 
                               grouping = peptide, 
                               log2_intensity = peptide_intensity_missing)
``` 

### Sample correlation

> Another way of quality control is to check the correlation of your samples. Ideally, replicates should cluster together and the different treatment conditions should be separated. We are now going to check if this is the case for our data by using the function `qc_sample_correlation()`. The function will return a correlation heatmap with a comparison of all samples.

Reviewer: ```suggestion
Another way of quality control is to check the correlation of your samples. Ideally, replicates should cluster together and different treatment conditions should be separated. We are now going to check if this is the case for our data by using the function `qc_sample_correlation()`. The function will return a correlation heatmap with a comparison of all samples.
```

```{r qc_sample_correlation, fig.width = 6, fig.height = 4, fig.align = "center"}
 qc_sample_correlation(data = data,
                       sample = sample, 
                       grouping = peptide, 
                       intensity = peptide_intensity_missing, 
                       condition = condition)
``` 

### Principal component analysis (PCA)

> Another popular quality control method (which could also be considered part of data analysis) is the principal component analysis (PCA). Generally, PCA is a method that reduces dimensionality of large datasets. In our case this helps us quickly assess how similar or different our replicates and conditions are. We are going to use `qc_pca()` to compute and plot a PCA for our data.

Reviewer: ```suggestion
Another popular quality control method (which could also be considered part of data analysis) is the principal component analysis (PCA). Generally, PCA is a method that reduces dimensionality of large datasets. In our case this helps us to quickly assess how similar or different our replicates and conditions are. We are going to use `qc_pca()` to compute and plot a PCA for our data.
```
